components/fxa-client/src/telemetry.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// A somewhat mixed-bag of all telemetry we want to collect. The idea is that
+// the app will "pull" telemetry via a new API whenever it thinks there might
+// be something to record.
+// It's considered a temporary solution until either we can record it directly
+// (eg, via glean) or we come up with something better.
+
+use serde_derive::*;
+
+// First, constituent parts
+#[derive(Debug, Serialize)]
+pub struct SentReceivedTab {
+    pub flow_id: String,
+    pub stream_id: String,
+}
+
+// We have a naive strategy to avoid unbounded memory growth - the intention
+// is that if any platform lets things grow to hit these limits, it's probably
+// never going to consume anything - so it doesn't matter what we discard (ie,
+// there's no good reason to have a smarter circular buffer etc)
+const MAX_TAB_EVENTS: usize = 20;
```

Comment:
This seems fairly low to me, and we definitely see instances of users sending more than 20 tabs in a single operation. The per-tab memory overhead here seems pretty low, what do you think about bumping this up by an order-of-magnitude or so?

Reply:
While Fenix makes that slightly more difficult (ie, no multi-device capability, and no multi-tab capability other than collections, I've no problem with that.

Reply:
> While Fenix makes that slightly more difficult (ie, no multi-device capability, and no multi-tab capability other than collections, I've no problem with that.

I mis-spoke - it does have "Send to all devices" - so it's going to be impossible to get desktop's flow semantics with the api we expose (ie, as mentioned above, we'll need an api which takes an array of device IDs) That probably shouldn't block getting the core version of this landed though.